Integrations/integration-Tenable_io.yml

```diff
@@ -0,0 +1,673 @@
+            sys.exit(0)
+        return res.json()
+
+    def get_scan_info(scans_result_elem):
+        response = send_scan_request(scans_result_elem['id'], ignore_license_error=True)
+        if response:
+            response['info'].update(scans_result_elem)
+            return response['info']
+
+    def send_vuln_details_request(plugin_id, date_range=None):
+        full_url = "{}{}{}/{}".format(BASE_URL, "workbenches/vulnerabilities/", plugin_id, "info")
+        res = requests.get(full_url, headers=AUTH_HEADERS, params=date_range_to_param(date_range))
+        return res.json()
+
+    def send_asset_vuln_request(asset_id, date_range):
+        full_url = "{}workbenches/assets/{}/vulnerabilities/".format(BASE_URL, asset_id)
+        res = requests.get(full_url, headers=AUTH_HEADERS, params=date_range_to_param(date_range))
+        res.raise_for_status()
+        return res.json()
+
+    def get_vuln_info(vulns):
+        vulns_info = {v['plugin_id']: v for v in vulns}
+        infos = []
+        errors = []
+        for pid, info in vulns_info.iteritems():
+            vuln_details = send_vuln_details_request(pid)
+            if u'error' in vuln_details:
+                errors.append(info)
+            else:
+                info.update(flatten(vuln_details['info']))
+                infos.append(info)
+        return infos, errors
+
+    def get_assets():
+        full_url = "{}{}".format(BASE_URL, "assets/")
+        res = requests.get(full_url, headers=AUTH_HEADERS)
+        return res.json()
+
+    def get_asset_id(hostname=None, ip=None):
+        assets = get_assets()
+        if 'error' in assets:
+            return None, { 'Type': entryTypes['error'], 'ContentsFormat': formats['text'], 'Contents': assets['error'] }
+        assets = assets['assets']
+        if hostname:
+            assets = filter(lambda x: hostname in x['fqdn'] + x['netbios_name'], assets)
+            if not assets:
+                return None, { 'Type': entryTypes['error'], 'ContentsFormat': formats['text'], 'Contents': 'No assets found for hostname: {}'.format(hostname) }
+            return assets[0]['id'], hostname, False
+        if ip:
+            assets = filter(lambda x: ip in x['ipv4'] + x['ipv6'], assets)
+            if not assets:
+                return None, { 'Type': entryTypes['error'], 'ContentsFormat': formats['text'], 'Contents': 'No assets found for ip: {}'.format(ip) }
+            return assets[0]['id'], ip, False
+
+    # Command methods
+    def test_module():
+        send_scan_request()
+        return 'ok'
+
+    def get_scans(folder_id, last_modification_date):
+        if last_modification_date:
+            last_modification_date = int(time.mktime(datetime.strptime(last_modification_date[0:len('YYYY-MM-DD')], "%Y-%m-%d").timetuple()))   # str(YYYY-MM-DD) to int(timestamp)
+        response = send_scan_request(folder_id=folder_id, last_modification_date=last_modification_date)
```

Comment:
we usually call the wrapper function
`get_scans_command` and the inner function `get_scans`
this way it is more clear.
please change all the functions accordingly

Reply:
The inner function is a wrapper for requests.request() and is used in many other functions, so `get_scans` is not a good name for it. There is also no need to wrap it in a `get_scans` function, because it will be a one-liner that is used only once.

I will change the "command" functions, but i'm keeping the name of the inner function, unless there is a better name that can be used in all calls to the function.